frontend-service/src/main/java/com/amazonaws/blox/frontend/operations/ListEnvironments.java

```diff
@@ -30,22 +36,29 @@
 
 @RestController
 public class ListEnvironments extends EnvironmentController {
+  @Autowired ListEnvironmentsMapper mapper;
 
   @RequestMapping(method = RequestMethod.GET)
   @ApiOperation(value = "List all environments")
   public ListEnvironmentsResponse listEnvironments(
       @PathVariable("cluster") String cluster,
       @RequestParam(value = "environmentNamePrefix", required = false) String environmentNamePrefix,
-      @ModelAttribute RequestPagination pagination) {
+      @ModelAttribute RequestPagination pagination)
+      throws ServiceException, ClientException {
 
-    return ListEnvironmentsResponse.builder().environmentNames(Collections.emptyList()).build();
+    // TODO: Add validation for cluster. Add Validation for environmentNamePrefix
```

Comment:
Do we have an issue to track adding external resource validation?

Reply:
I've created one.